lambdaclass/mirra_backend
# [GH-404] lowest and highest stat targeting strategies

## Motivation

We need to implement two new targeting strategies, that will target the unit with lowest or highest value of a stat, depending the case.

Closes https://github.com/lambdaclass/afk_gacha_game/issues/404

## Summary of changes

- Implement lowest stat strategy
- Implement highest stat strategy
- Add tests

## How to test it?

1. Start the client in the main branch, and the backend with this branch.
2. Pick one of your units and change its basic skill's targeting strategy to lowest. You should end up with something like the following:
```
"targeting_strategy": {
     "type": {"lowest": "health"},
     "target_allies": false
}
```
3. Run `Champions.Config.import_skill_config` in your iex console to apply the changes.
4. Run a battle in the client. To make things easier, you can only leave the unit you modified in your line up.
   4.1. Remember to change all `Logger.info()` to `IO.inspect()`, so that you can see the complete log in your terminal.
5. Check the logs, and verify that the unit you changed is always attacking the opponent's unit with the lowest health.
6. Repeat to test the case for `highest` strategy.

Also, you can check the tests that were added in this PR

## Checklist
